### DISCUSSION TITLE
keep speed - different function than before

### DISCUSSION THREAD

**Discussion by ElfVierzehn** [Mogli12/KeyboardSteer#809] [General]
Hey there!

I just updated your beautiful mod and as a mouse and keyboard player, I use the keep speed function while holding a key. Until now it kept the speed exactly where it was when pressing my specific key for 'keep speed'.

After updating, 'keep speed' now automatically accelerates to a certain speed. So when I brake to go around a narrow corner with like 5 kph I press the keep speed button, it does not stay ar 5 kph, but accelerates to 15 kph. Why is that and can I change that?

Thanks for any input!

**Comment by Mogli12** [comment]
Hello,

It is difficult to guess from which version to which version to did upgrade.

I assume that you have a vehicle with manual gearbox. In this case the lowest possible keep speed is the speed of that gear at idle RPM. That was already the keep speed implementation in FS19.

**Comment by ElfVierzehn** [comment]
I am not very certain which the old version was, I downloaded the new version 5 minutes before I wrote this topic.

Used it in FS19 as well, it never did this, although I used various manual gearboxes. When ever I pressed the keep speed button it kept exactly the speed I was driving. Now the tractor is aaccelerating.

It's my first savegame in FS22 and I use the machines for over 100 hours each so I got quite used to it. As I continued today I immediately felt something is off and took a while to realize what it was. Especially when backing up with very low speed I used the keep speed function and it never just accelerated.

Next to this mod I just use Giants mods and CP. Didn't change anything and the log also shows no error (which I wouldn't think with that kind of behavior in game anyway).

Thank you very much for your reply, seems like I have to get used to it. I do understand why it does that and it's only logical at this point - it still never appeared before. Thanks again!
